src/components/UIShell/SideNav.js

```diff
@@ -5,131 +5,120 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import React, { useState, useRef } from 'react';
 import { settings } from 'carbon-components';
 import cx from 'classnames';
-import React from 'react';
 import PropTypes from 'prop-types';
 import { AriaLabelPropType } from '../../prop-types/AriaPropTypes';
 import SideNavFooter from './SideNavFooter';
 
 const { prefix } = settings;
 
-const translations = {
-  'carbon.sidenav.state.open': 'Close',
-  'carbon.sidenav.state.closed': 'Open',
-};
-
-function translateById(id) {
-  return translations[id];
-}
-
-export default class SideNav extends React.Component {
-  static propTypes = {
-    /**
-     * Specify whether the side navigation is expanded or collapsed
-     */
-    isExpanded: PropTypes.bool,
-    /**
-     * Required props for accessibility label on the underlying menu
-     */
-    ...AriaLabelPropType,
-
-    /**
-     * Optionally provide a custom class to apply to the underlying <li> node
-     */
-    className: PropTypes.string,
-
-    /**
-     * Provide a custom function for translating all message ids within this
-     * component. This function will take in two arguments: the mesasge Id and the
-     * state of the component. From this, you should return a string representing
-     * the label you want displayed or read by screen readers.
-     */
-    translateById: PropTypes.func,
+const SideNav = React.forwardRef(function SideNav(props, ref) {
+  const {
+    expanded: expandedProp,
+    defaultExpanded,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+    children,
+    onToggle,
+    className: customClassName,
+    translateById: t,
+  } = props;
+  const { current: controlled } = useRef(expandedProp != null);
```

Comment:
should we use strict inequality check here?

Reply:
That's a good question, I assume you mean `!== undefined`? I think you're right since our stance is "using this prop means you control the component."

Reply:
I initially just meant the `!==` operator but yeah I guess it does make sense to look for `undefined`

Reply:
Yeah, I think !== null would miss the main case we want to identify (they don't provide any prop)